src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -633,16 +636,31 @@ export class ResultsViewPageStore {
         }
     });
 
-    @computed get geneToMutationData():{[hugoGeneSymbol:string]:CacheData<Mutation[]>|null} {
-        if (this.genes.result) {
-            return this.genes.result.reduce((map:{[hugoGeneSymbol:string]:CacheData<Mutation[]>|null}, gene:Gene)=>{
-                map[gene.hugoGeneSymbol] = this.mutationDataCache.get({ entrezGeneId: gene.entrezGeneId });
-                return map;
-            }, {});
-        } else {
-            return {};
+    readonly geneToMutationData = remoteData<{[hugoGeneSymbol:string]:CacheData<Mutation[]>|null}>({
+        await:()=>[
+            this.genes,
+            this.mutationDataCache
+        ],
+        invoke:()=>{
+            return this.mutationDataCache.result!.awaitComplete(this.genes.result!, true).then(()=>{
```

Comment:
heres where the action is @alisman

Reply:
so the second argument says whether you want to passively watch, or watch and also ask for the data to be loaded

Reply:
What's use case for passively watching?

Reply:
not sure of one at the moment but it doesnt seem obviously useless to me

Reply:
actually I think you're right